### ISSUE TITLE
[Feature Request] Assign groups for special events

### ISSUE THREAD

**Issue by mightyBroccoli** [Newcomer1989/TSN-Ranksystem#231]
We have a funny tradition to assign groups on special events eg "Whale you be my Valentine?" if the specific user is connected on valentines day or "Christmas angel" when the specific user is connected on christmas.

On a more general level define a date and a group to assign on that date. Perhaps even more eloquent to assign the duration of the group 1 year or what ever.

What do you guys think about that?

**Comment by bykidi** [comment]
Memorable group for those who spend %customspenttime% hours during specific day on the server?

**Comment by bykidi** [comment]
Idea of mine: own badge system. Badges could be available for reaching online time during specific periods (like making the online time on the server during the holidays etc etc)

**Comment by mightyBroccoli** [comment]
@bykidi generally your idea is cool with my request except one point that the users need to self assign thier granted groups, normally ppl are lazy and don't want to bother with that stuff.
That's why my request is about automatically getting the special group in that time frame or date.

**Comment by bykidi** [comment]
@mightyBroccoli agreed with you, but i think we should get both - auto-assign and ability to remove&assign again for users themselfs

**Comment by Shad86** [comment]
Im also more for the "automatic thing"

like said:

Assign groups on a specific time on a specific event (for example Christmas, Valentines day, and so on...) for a specific time - after the event it gets removed.

Why i think this is the best?

because you will run into troubles if you get more and more events.

Im also not a fan of teamspeak servers where people have like "50" Icons

At the end you only see icons and thats very unclear and is in my opinion very unhealthy for a good community.

Maybe as a "permanent solution" an option like set one group for participate on Events like:

"has participate on one or more Events 2017" would be a good solution because then you didn't have so much "icon spam" in the tree

**Comment by Newcomer1989** [comment]
This what you described here is a nice idea. I see this as an own project like "TS eventsystem". But you really think this should be a part of the Ranksystem?

**Comment by mightyBroccoli** [comment]
> because you will run into troubles if you get more and more events.
> Im also not a fan of teamspeak servers where people have like "50" Icons

@Shad86 I totally agree with you with the massive amount of emblems, except one thing. The whole idea is gone when the only thing is "was there 20XX" The main idea is to let somebody know who was there Christmas or whatever not generally was there 20XX. I think that would totally miss the intention.


> But you really think this should be a part of the Ranksystem?

@Newcomer1989 Yes I totally think that would be appropriate for the ranksystem. Yes the first thought for a ranksystem is something more like Level/ranksystem to "level up" the user. In my opinion this idea is more like a different layer to this "Level System" and total appropriate to make the leveling up experience complete.

**Comment by SpliinTaX** [comment]
Könnte zum Beispiel auch nützlich sein für sogenannte Happy Hours wo man schneller levelt etc. Das alle Leute automatisch die Gruppe zu gewiesen bekommen.

**Comment by Newcomer1989** [comment]
OK, you convinced me ;-)
It will be added in a simplified version in one of the next releases.
